2006-10-21  Guillem Jover  <guillem@hadrons.org>

	* syslogd/syslogd.c (init): Close CF properly before returning.


Index: b/syslogd/syslogd.c
===================================================================
--- a/syslogd/syslogd.c
+++ b/syslogd/syslogd.c
@@ -1657,6 +1657,7 @@ init (int signo)
     {
       /* There is no graceful recovery here.  */
       dbg_printf ("cannot allocate space for configuration\n");
+      fclose (cf);
       return;
     }
   cline = cbuf;
@@ -1696,6 +1697,7 @@ init (int signo)
 	    {
 	      /* Sigh ...  */
 	      dbg_printf ("cannot allocate space configuration\n");
+	      fclose (cf);
 	      free (cbuf);
 	      return;
 	    }
